### PR TITLE
Add set_system_time function

### LINKED ISSUE #31: set_system_time is not supported
The original temporal_tables extension defined set_system_time:

https://github.com/arkhipov/temporal_tables#advanced-usage

I cannot use my code without it!  Any chance of adding this simulation of the original as well?

### PR DESCRIPTION
**Pull request by radomird**
Closes #31 

I've added a new file with `set_system_time` function that will set a configuration parameter on the db called `user_defined.system_time` and fill in the desired value. The `versioning` function will then try to use this config parameter to set the `time_stamp_to_use`.

- added new function ✅ 
- updated `versioning_function` and `versioning_function_no_checks` too use the new config parameter ✅ 
- updated README ✅ 
- add tests ✅

### CODE REVIEW

**Comment by radomird** [comment]
Hi @paolochiodi I've tried to implement the requested missing feature. Could you please take a look if I'm heading in the right direction?

Thanks.

**Comment by simoneb** [comment]
@paolochiodi can you please take a look at this PR? we forgot about it

**Comment by dzolo** [comment]
Nice missing feature. I would suggest to use datetime format `YYYY-MM-DD HH24:MI:SS.US` in order to be capable to set precise time.

It would be also nice to have timezone support (as in original support) but this will be more work to be done as `TO_TIMESTAMP` does not support TZ. Maybe cast to `timestamp with time zone` could be used instead of `TO_TIMESTAMP`?

**Comment by radomird** [comment]
@dzolo adding ms portion to the timestamp is not a bad idea. I will update the date/time format.

Regarding the Timezone support - would be a nice addition, however I would probably open a feature request issue and that can be tackled in a separate PR, as it is a bit more complex to implement.

**Comment by radomird** [comment]
Support for milliseconds and microseconds added. They are both optional. I've also bumped up the version in the `package.json` to `0.6.0`.

@dzolo a review of the PR would be appreciated 😉

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "temporal_tables",
-  "version": "0.5.0",
+  "version": "0.6.0",
```
Comment (simoneb) [inline package.json:3]: don't bump the version, this is now automated

File: versioning_function.sql [hunk 2]
```diff
@@ -1,4 +1,4 @@
--- version 0.5.0
+-- version 0.6.0
```
Comment (simoneb) [inline versioning_function.sql:1]: same here

File: package.json (hunk 1 above)
Comment (radomird) [reply package.json]: Thanks, reverted.

File: README.md [hunk 3]
```diff
@@ -25,6 +23,12 @@ Create a database and the versioning function:
 ```sh
 createdb temporal_test
 psql temporal_test < versioning_function.sql
+psql temporal_test < system_time_function.sql
+```
+
+If you would like to have `set_system_time` function available (more details [below](#system-time)) you should run the following as well:
+```sh
+psql temporal_test < system_time_function.sql
```
Comment (simoneb) [inline README.md:31]: I don't understand this passage. Why are we duplicating the instruction?

**Review by simoneb — APPROVED** [review]
a couple of comments but overall LGTM, nice job 👌

File: README.md (hunk 3 above)
Comment (radomird) [reply README.md]: Whoops, it seems like a copy/paste issue. Fixed.

**Pull request merged**